### PR TITLE
Add dev docs for checkboxes

### PR DESCRIPTION
**Pull request by perclasson**
Add dev docs for checkboxes

### CODE REVIEW

File: docs/components/checkboxes.md [hunk 1]
```diff
@@ -0,0 +1,225 @@
...
+
+- Select one or more options from a list
+- Present a list containing sub-selections
+- Turn an item on or off in a desktop environment
+
+![Checkbox hero example for menu options](assets/checkboxes/checkbox-hero.png)
+
+## Contents
+
+- [Using checkboxes](#using-checkboxes)
```
Comment (guidezpl) [inline docs/components/checkboxes.md:24]: Add a contents link and one for all 2-level headers

Comment (perclasson) [reply docs/components/checkboxes.md:25]: Done!

File: docs/components/checkboxes.md [hunk 2]
```diff
@@ -0,0 +1,226 @@
...
+            onChanged: i == 4 ? null : (bool value) {
+              setState(() {
+                checked[i] = value;
+              });
+            },
+            tristate: i == 1,
+            value: checked[i],
+            activeColor: Color(0xFF6200EE),
+          ),
+          Text('Checkbox ${i + 1}')
```
Comment (rami-a) [inline docs/components/checkboxes.md:67]: In my radio button docs, I also change the text color for the disabled row, I think that may look a little better:

https://github.com/material-components/material-components-flutter/blob/master/docs/components/radio_buttons.md

I suggest doing it for all the examples and the dartpad code

Comment (perclasson) [reply docs/components/checkboxes.md:67]: That makes sense. Updating now.

**Pull request merged**